src/barplot.ts

```diff
@@ -247,6 +247,15 @@ export function drawLabels(size:number[], data:INumericalVector, labels: d3.Sele
     'display': (rowHeight >= 15) ? 'inline' : 'none',
     'font-size': '14px'
   });
+  const categories = (<any>data.desc.value).categories;
+  const nameToLabel = new Map<string, string>();
+  if (categories) {
```

Comment:
Well, the argument typing is wrong, it should be a generic vector: `IAnyVector` since it can be an `ICategoricalVector`, too